File: client/lib/api/v2/schemas/pricing.ts
/**
 * Pricing Analytics Schemas
 */

import { z } from "zod";

export const DebugInfoItemSchema = z.object({
  id: z.string(),
  created_at: z.string(),
  content: z.string(),
});

export const ModelRunItemSchema = z.object({
  model_run_id: z.string(),
  created_at: z.string(),
  input_tokens: z.number(),
  output_tokens: z.number(),
  model_id: z.string().nullable(),
  profile_id: z.string().nullable(),
  agent_id: z.string().nullable(),
  persona_id: z.string().nullable(),
  debug_info: z.array(DebugInfoItemSchema),
});

export const ModelMappingWithPricingSchema = z.object({
  name: z.string(),
  description: z.string(),
  input_ppm: z.number(),
  output_ppm: z.number(),
});

export const PricingAnalyticsResponseSchema = z.object({
  model_runs: z.array(ModelRunItemSchema),
  model_mapping: z.record(z.string(), ModelMappingWithPricingSchema),
  profile_mapping: z.record(z.string(), z.string()),
  agent_mapping: z.record(z.string(), z.string()),
  persona_mapping: z.record(z.string(), z.string()),
});

export type DebugInfoItem = z.infer<typeof DebugInfoItemSchema>;
export type ModelRunItem = z.infer<typeof ModelRunItemSchema>;
export type ModelMappingWithPricing = z.infer<
  typeof ModelMappingWithPricingSchema
>;
export type PricingAnalyticsResponse = z.infer<
  typeof PricingAnalyticsResponseSchema
>;

File: server/app/schemas/pricing.py
"""Pricing analytics schemas."""

from pydantic import BaseModel


class DebugInfoItem(BaseModel):
    """Debug information item."""

    id: str
    created_at: str
    content: str


class ModelRunItem(BaseModel):
    """Model run item with aggregated metrics."""

    model_run_id: str
    created_at: str
    input_tokens: int
    output_tokens: int
    model_id: str | None = None
    profile_id: str | None = None
    agent_id: str | None = None
    persona_id: str | None = None
    debug_info: list[DebugInfoItem] | None = None


class ModelMappingWithPricing(BaseModel):
    """Model mapping with pricing information."""

    name: str
    description: str
    input_ppm: float
    output_ppm: float


class PricingAnalyticsResponse(BaseModel):
    """Response for pricing analytics."""

    model_runs: list[ModelRunItem]
    model_mapping: dict[str, ModelMappingWithPricing]
    profile_mapping: dict[str, str]
    agent_mapping: dict[str, str]
    persona_mapping: dict[str, str]
